Apps/W1/HybridGP/app/src/pages/GPMigrationConfiguration.Page.al

```diff
@@ -0,0 +1,512 @@
+        ShowManagementPromptOnClose := true;
+    end;
+
+    trigger OnOpenPage()
+    var
+        IntroNotification: Notification;
+    begin
+        if ShowIntroductionNotification then begin
+            IntroNotification.Message(IntroNotificationMsg);
+            IntroNotification.Send();
+        end;
+
+        if not Rec.Get() then
+            Rec.Insert();
+
+        CurrPage.SetRecord(Rec);
+        EnsureSettingsForAllCompanies();
+    end;
+
+    local procedure EnsureSettingsForAllCompanies()
+    var
+        GPCompanyAdditionalSettingsEachCompany: Record "GP Company Additional Settings";
+        HybridCompany: Record "Hybrid Company";
+    begin
+        HybridCompany.SetRange(Replicate, true);
+        if HybridCompany.FindSet() then begin
+            repeat
+                if not GPCompanyAdditionalSettingsEachCompany.Get(HybridCompany.Name) then begin
+                    GPCompanyAdditionalSettingsEachCompany.Validate(Name, HybridCompany.Name);
+                    GPCompanyAdditionalSettingsEachCompany.Validate("Migrate Inactive Customers", Rec."Migrate Inactive Customers");
+                    GPCompanyAdditionalSettingsEachCompany.Validate("Migrate Inactive Vendors", Rec."Migrate Inactive Vendors");
+                    GPCompanyAdditionalSettingsEachCompany.Validate("Migrate Inactive Checkbooks", Rec."Migrate Inactive Checkbooks");
+                    GPCompanyAdditionalSettingsEachCompany.Validate("Migrate Vendor Classes", Rec."Migrate Vendor Classes");
+                    GPCompanyAdditionalSettingsEachCompany.Validate("Migrate Customer Classes", Rec."Migrate Customer Classes");
+                    GPCompanyAdditionalSettingsEachCompany.Validate("Migrate Item Classes", Rec."Migrate Item Classes");
+                    GPCompanyAdditionalSettingsEachCompany.Validate("Migrate Bank Module", Rec."Migrate Bank Module");
+                    GPCompanyAdditionalSettingsEachCompany.Validate("Migrate Payables Module", Rec."Migrate Payables Module");
+                    GPCompanyAdditionalSettingsEachCompany.Validate("Migrate Receivables Module", Rec."Migrate Receivables Module");
+                    GPCompanyAdditionalSettingsEachCompany.Validate("Migrate Open POs", Rec."Migrate Open POs");
+                    GPCompanyAdditionalSettingsEachCompany.Validate("Migrate Inventory Module", Rec."Migrate Inventory Module");
+                    GPCompanyAdditionalSettingsEachCompany.Validate("Oldest GL Year To Migrate", Rec."Oldest GL Year To Migrate");
+                    GPCompanyAdditionalSettingsEachCompany.Insert();
+                end;
+            until HybridCompany.Next() = 0;
+        end;
+
+        CurrPage.Update();
+    end;
+
+    local procedure PrepSettingsForFieldUpdate()
+    begin
+        GPCompanyAdditionalSettings.SetFilter("Name", '<>%1', '');
+        GPCompanyAdditionalSettings.FindSet();
+    end;
+
+    local procedure AfterFieldUpdate()
+    begin
+        CurrPage.Update();
```

Comment:
Is this really needed? CurrPage.Update() is a bit expensive and may not be needed.

Reply:
Looks like not now, an earlier implementation of this did need it though.